golem-templates/templates/python/python-app-component-example-gitpulse-agent/components-python/component-name/src/component.py

```diff
@@ -0,0 +1,399 @@
+    temperature=0.2,
+    tools=[],
+    provider_options=[],
+    max_tokens=None,
+    tool_choice=None,
+    stop_sequences=None,
+)
+
+
+def get_state() -> State:
+    if state is None:
+        raise ValueError("Worker is not initialized")
+    return state
+
+
+class ComponentNameApi(exports.ComponentNameApi):
+    def initialize(self, args: InitializeArgs) -> None:
+        global state
+        state = State(
+            last_checked=None,
+            pending_notifications=[],
+            seen_notifications=set(),
+            github_username=args.github_username,
+            github_auth_token=args.github_auth_token,
+        )
+
+    def get_last_checked(self) -> Optional[str]:
+        return get_state().last_checked
+
+    def get_whats_new(self) -> WhatsNewResult:
+        last_checked = update_notifications()
+        result = WhatsNewResult(
+            action_needed=[], fyi=[], releases=[], ci_cd=[], last_checked=last_checked
+        )
+        for notification in get_state().pending_notifications:
+            if notification.category == Category.ACTION_NEEDED:
+                result.action_needed.append(notification)
+            elif notification.category == Category.FYI:
+                result.fyi.append(notification)
+            elif notification.category == Category.RELEASES:
+                result.releases.append(notification)
+            else:
+                result.ci_cd.append(notification)
+        return result
+
+    def has_seen_notification(self, id: str) -> bool:
+        return id in get_state().seen_notifications
+
+    def mark_all_seen(self) -> None:
+        state = get_state()
+        for notification in state.pending_notifications:
+            state.seen_notifications.add(notification.id)
+        state.pending_notifications.clear()
+
+    def mark_notification_seen(self, id: str) -> None:
+        state = get_state()
+        state.seen_notifications.add(id)
+        filtered_notifications = list(
+            filter(lambda n: n.id != id, state.pending_notifications)
+        )
+        state.pending_notifications = filtered_notifications
+
+
+def update_notifications() -> str:
+    state = get_state()
+    since = state.last_checked
+    before = datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    new_notifications = get_notifications(since, before)
+
+    results: list[EnhancedNotification] = []
+    for notification in new_notifications:
+        notification_subject = notification["subject"]["type"]
+        if notification_subject == "Issue":
+            results.append(process_issue_notification(since, before, notification))
+        elif notification_subject == "PullRequest":
+            results.append(process_issue_notification(since, before, notification))
+        elif notification_subject == "CheckSuite":
+            results.append(process_check_suite_notification(notification))
+        else:
+            # TODO: implement other notification types
```

Comment:
i think we should not leave TODOs in the template

Reply:
even if they are true :D

Reply:
Better if I call them Optional:? :D

Reply:
sorry, exactly what wanted to ask, to make them simple guide comments, thanky you